tests/test_oml/test_metrics/test_embedding_metrics.py

```diff
@@ -126,33 +111,30 @@ def run_retrieval_metrics(case) -> None:  # type: ignore
     num_samples = len(dataset)
     calc = EmbeddingMetrics(
         dataset=dataset,
-        embeddings_key=dataset.input_tensors_key,
-        labels_key=LABELS_KEY,
-        is_query_key=IS_QUERY_KEY,
-        is_gallery_key=IS_GALLERY_KEY,
-        categories_key=CATEGORIES_KEY,
         cmc_top_k=top_k,
         precision_top_k=tuple(),
         map_top_k=tuple(),
         fmr_vals=tuple(),
         pcf_variance=tuple(),
-        postprocessor=get_trivial_postprocessor(top_n=2),
+        postprocessor=get_trivial_postprocessor(top_n=num_samples),
     )
 
-    calc.setup(num_samples=num_samples)
+    calc.setup()
 
     for batch in DataLoader(dataset, batch_size=4, shuffle=False):
-        calc.update_data(batch)
+        calc.update(embeddings=batch[dataset.input_tensors_key], indices=batch[dataset.index_key])
 
     metrics = calc.compute_metrics()
 
     compare_dicts_recursively(gt_metrics, metrics)
 
     # the euclidean distance between any one-hots is always sqrt(2) or 0
-    assert compare_tensors_as_sets(calc.distance_matrix, torch.tensor([0, math.sqrt(2)]))
+    assert (
+        torch.isclose(calc.retrieval_results.distances, torch.tensor([0.0])).any()
```

Comment:
the logic has changed a bit because now we only have distances to the closest items, and don't have access to the whole distance matrix

thus, we cannot guarantee we meet all the potential distances values as before, but what we actually meet must much these potential values